Synthesis failed for google-http-java-client

Hello! Autosynth couldn't regenerate google-http-java-client. :broken_heart:

Here's the output from running `synth.py`:

```
   return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/google-http-java-client/synth.py", line 21, in <module>
    templates = common_templates.java_library()
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 95, in java_library
    return self._generic_library("java_library", **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 57, in _generic_library
    result = t.render(**kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 83, in render
    _render_to_path(self.env, template_name, self.dir / subdir, kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 53, in _render_to_path
    output.dump(fh)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1313, in dump
    fp.writelines(iterable)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1357, in __next__
    return self._next()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1125, in generate
    yield self.environment.handle_exception()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/home/kbuilder/.cache/synthtool/synthtool/synthtool/gcp/templates/java_library/.github/workflows/ci.yaml", line 12, in top-level template code
    java: [{% if metadata['min_java_version'] <= 7 %}7, {% endif %}8, 11]
jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'min_java_version'
2020-06-19 13:26:21,282 autosynth [ERROR] > Synthesis failed
2020-06-19 13:26:21,288 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at a82b980 chore(deps): update dependency org.apache.maven.plugins:maven-project-info-reports-plugin to v3.1.0 (#1050)
2020-06-19 13:26:21,295 autosynth [DEBUG] > Running: git checkout autosynth-synthtool
Switched to branch 'autosynth-synthtool'
2020-06-19 13:26:21,302 autosynth [ERROR] > Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.
2020-06-19 13:26:21,443 autosynth [DEBUG] > Running: git checkout a82b9808f112c240c213726356e610e3a9fb4e61
Note: checking out 'a82b9808f112c240c213726356e610e3a9fb4e61'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at a82b980 chore(deps): update dependency org.apache.maven.plugins:maven-project-info-reports-plugin to v3.1.0 (#1050)
2020-06-19 13:26:21,450 autosynth [DEBUG] > Running: git checkout 3e6fed750653a8f41a5efeca0bc16b756a9532bb
HEAD is now at 3e6fed7 Revert "ci(java): run dependency test on Java 8 and 11 (#633)" (#636)
2020-06-19 13:26:21,459 autosynth [DEBUG] > Running: git branch -f autosynth-70
2020-06-19 13:26:21,463 autosynth [DEBUG] > Running: git checkout autosynth-70
Switched to branch 'autosynth-70'
2020-06-19 13:26:21,469 autosynth [INFO] > Running synthtool
2020-06-19 13:26:21,470 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']
2020-06-19 13:26:21,470 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/google-http-java-client/70/sponge_log.log
2020-06-19 13:26:21,472 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata synth.metadata synth.py --
2020-06-19 13:26:21,679 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/google-http-java-client/synth.py.
On branch autosynth-70
nothing to commit, working tree clean
2020-06-19 13:26:21,813 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
DEBUG:synthtool:Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
.github/CODEOWNERS
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/release-please.yml
.github/trusted-contribution.yml
.github/workflows/ci.yaml
2020-06-19 13:26:21,835 synthtool [DEBUG] > Wrote metadata to synth.metadata.
DEBUG:synthtool:Wrote metadata to synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/google-http-java-client/synth.py", line 21, in <module>
    templates = common_templates.java_library()
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 95, in java_library
    return self._generic_library("java_library", **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 57, in _generic_library
    result = t.render(**kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 83, in render
    _render_to_path(self.env, template_name, self.dir / subdir, kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 53, in _render_to_path
    output.dump(fh)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1313, in dump
    fp.writelines(iterable)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1357, in __next__
    return self._next()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1125, in generate
    yield self.environment.handle_exception()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/home/kbuilder/.cache/synthtool/synthtool/synthtool/gcp/templates/java_library/.github/workflows/ci.yaml", line 12, in top-level template code
    java: [{% if metadata['min_java_version'] <= 7 %}7, {% endif %}8, 11]
jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'min_java_version'
2020-06-19 13:26:21,879 autosynth [ERROR] > Synthesis failed
2020-06-19 13:26:21,885 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at a82b980 chore(deps): update dependency org.apache.maven.plugins:maven-project-info-reports-plugin to v3.1.0 (#1050)
2020-06-19 13:26:21,893 autosynth [DEBUG] > Running: git checkout autosynth
Switched to branch 'autosynth'
2020-06-19 13:26:21,900 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 649, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 498, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 629, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 393, in synthesize_loop
    synthesize_inner_loop(toolbox, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 403, in synthesize_inner_loop
    synthesizer, len(toolbox.versions) - 1
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 275, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/51b19127-021c-490f-9add-0c6fb0ed82bf/targets/github%2Fsynthtool;config=default/tests;query=google-http-java-client;failed=false).

Autosynth is still having trouble generating google-http-java-client. :sob:

Here's the output from running `synth.py`:

```
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/google-http-java-client/synth.py", line 21, in <module>
    templates = common_templates.java_library()
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 95, in java_library
    return self._generic_library("java_library", **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 57, in _generic_library
    result = t.render(**kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 83, in render
    _render_to_path(self.env, template_name, self.dir / subdir, kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 53, in _render_to_path
    output.dump(fh)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1313, in dump
    fp.writelines(iterable)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1357, in __next__
    return self._next()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1125, in generate
    yield self.environment.handle_exception()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/home/kbuilder/.cache/synthtool/synthtool/synthtool/gcp/templates/java_library/.github/workflows/ci.yaml", line 12, in top-level template code
    java: [{% if metadata['min_java_version'] <= 7 %}7, {% endif %}8, 11]
jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'min_java_version'
2020-06-19 14:34:20,481 autosynth [ERROR] > Synthesis failed
2020-06-19 14:34:20,481 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at a82b980 chore(deps): update dependency org.apache.maven.plugins:maven-project-info-reports-plugin to v3.1.0 (#1050)
2020-06-19 14:34:20,488 autosynth [DEBUG] > Running: git checkout autosynth-synthtool
Switched to branch 'autosynth-synthtool'
2020-06-19 14:34:20,493 autosynth [ERROR] > Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.
2020-06-19 14:34:20,619 autosynth [DEBUG] > Running: git checkout a82b9808f112c240c213726356e610e3a9fb4e61
Note: checking out 'a82b9808f112c240c213726356e610e3a9fb4e61'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at a82b980 chore(deps): update dependency org.apache.maven.plugins:maven-project-info-reports-plugin to v3.1.0 (#1050)
2020-06-19 14:34:20,626 autosynth [DEBUG] > Running: git checkout 483f18635c3af10031435535268abe001ad69e7c
HEAD is now at 483f186 chore: assign code ownership of template directories to language teams (#634)
2020-06-19 14:34:20,631 autosynth [DEBUG] > Running: git branch -f autosynth-72
2020-06-19 14:34:20,634 autosynth [DEBUG] > Running: git checkout autosynth-72
Switched to branch 'autosynth-72'
2020-06-19 14:34:20,639 autosynth [INFO] > Running synthtool
2020-06-19 14:34:20,639 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']
2020-06-19 14:34:20,639 autosynth [DEBUG] > log_file_path: /tmpfs/src/github/synthtool/logs/googleapis/google-http-java-client/72/sponge_log.log
2020-06-19 14:34:20,641 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata synth.metadata synth.py --
2020-06-19 14:34:20,853 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/google-http-java-client/synth.py.
On branch autosynth-72
nothing to commit, working tree clean
2020-06-19 14:34:20,987 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
DEBUG:synthtool:Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
.github/CODEOWNERS
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/release-please.yml
.github/trusted-contribution.yml
.github/workflows/ci.yaml
2020-06-19 14:34:21,009 synthtool [DEBUG] > Wrote metadata to synth.metadata.
DEBUG:synthtool:Wrote metadata to synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/google-http-java-client/synth.py", line 21, in <module>
    templates = common_templates.java_library()
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 95, in java_library
    return self._generic_library("java_library", **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 57, in _generic_library
    result = t.render(**kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 83, in render
    _render_to_path(self.env, template_name, self.dir / subdir, kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 53, in _render_to_path
    output.dump(fh)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1313, in dump
    fp.writelines(iterable)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1357, in __next__
    return self._next()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1125, in generate
    yield self.environment.handle_exception()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/home/kbuilder/.cache/synthtool/synthtool/synthtool/gcp/templates/java_library/.github/workflows/ci.yaml", line 12, in top-level template code
    java: [{% if metadata['min_java_version'] <= 7 %}7, {% endif %}8, 11]
jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'min_java_version'
2020-06-19 14:34:21,043 autosynth [ERROR] > Synthesis failed
2020-06-19 14:34:21,043 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at a82b980 chore(deps): update dependency org.apache.maven.plugins:maven-project-info-reports-plugin to v3.1.0 (#1050)
2020-06-19 14:34:21,050 autosynth [DEBUG] > Running: git checkout autosynth
Switched to branch 'autosynth'
2020-06-19 14:34:21,057 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 625, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 482, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 605, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 377, in synthesize_loop
    synthesize_inner_loop(toolbox, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 387, in synthesize_inner_loop
    synthesizer, len(toolbox.versions) - 1
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 266, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/2e45955d-4978-4bbf-b28a-41cdec188631/targets/github%2Fsynthtool;config=default/tests;query=google-http-java-client;failed=false).

Autosynth is still having trouble generating google-http-java-client. :sob:

Here's the output from running `synth.py`:

```
mpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1357, in __next__
    return self._next()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1125, in generate
    yield self.environment.handle_exception()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/home/kbuilder/.cache/synthtool/synthtool/synthtool/gcp/templates/java_library/.github/workflows/ci.yaml", line 12, in top-level template code
    java: [{% if metadata['min_java_version'] <= 7 %}7, {% endif %}8, 11]
jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'min_java_version'
2020-06-20 01:34:18,504 autosynth [ERROR] > Synthesis failed
2020-06-20 01:34:18,504 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at a82b980 chore(deps): update dependency org.apache.maven.plugins:maven-project-info-reports-plugin to v3.1.0 (#1050)
2020-06-20 01:34:18,511 autosynth [DEBUG] > Running: git checkout autosynth-synthtool
Switched to branch 'autosynth-synthtool'
2020-06-20 01:34:18,517 autosynth [ERROR] > Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.
2020-06-20 01:34:18,641 autosynth [DEBUG] > Running: git checkout a82b9808f112c240c213726356e610e3a9fb4e61
Note: checking out 'a82b9808f112c240c213726356e610e3a9fb4e61'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at a82b980 chore(deps): update dependency org.apache.maven.plugins:maven-project-info-reports-plugin to v3.1.0 (#1050)
2020-06-20 01:34:18,648 autosynth [DEBUG] > Running: git checkout 4f2c9f752a94042472fc03c5bd9e06e89817d2bd
HEAD is now at 4f2c9f7 ci(java): run dependency test on Java 8 and 11 (#639)
2020-06-20 01:34:18,653 autosynth [DEBUG] > Running: git branch -f autosynth-74
2020-06-20 01:34:18,657 autosynth [DEBUG] > Running: git checkout autosynth-74
Switched to branch 'autosynth-74'
2020-06-20 01:34:18,662 autosynth [INFO] > Running synthtool
2020-06-20 01:34:18,663 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']
2020-06-20 01:34:18,663 autosynth [DEBUG] > log_file_path: /tmpfs/src/github/synthtool/logs/googleapis/google-http-java-client/74/sponge_log.log
2020-06-20 01:34:18,665 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata synth.metadata synth.py --
2020-06-20 01:34:18,867 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/google-http-java-client/synth.py.
On branch autosynth-74
nothing to commit, working tree clean
2020-06-20 01:34:18,999 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
DEBUG:synthtool:Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
.github/CODEOWNERS
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/release-please.yml
.github/trusted-contribution.yml
.github/workflows/ci.yaml
2020-06-20 01:34:19,024 synthtool [DEBUG] > Wrote metadata to synth.metadata.
DEBUG:synthtool:Wrote metadata to synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/google-http-java-client/synth.py", line 21, in <module>
    templates = common_templates.java_library()
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 95, in java_library
    return self._generic_library("java_library", **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 57, in _generic_library
    result = t.render(**kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 83, in render
    _render_to_path(self.env, template_name, self.dir / subdir, kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 53, in _render_to_path
    output.dump(fh)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1313, in dump
    fp.writelines(iterable)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1357, in __next__
    return self._next()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1125, in generate
    yield self.environment.handle_exception()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/home/kbuilder/.cache/synthtool/synthtool/synthtool/gcp/templates/java_library/.github/workflows/ci.yaml", line 12, in top-level template code
    java: [{% if metadata['min_java_version'] <= 7 %}7, {% endif %}8, 11]
jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'min_java_version'
2020-06-20 01:34:19,062 autosynth [ERROR] > Synthesis failed
2020-06-20 01:34:19,063 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at a82b980 chore(deps): update dependency org.apache.maven.plugins:maven-project-info-reports-plugin to v3.1.0 (#1050)
2020-06-20 01:34:19,070 autosynth [DEBUG] > Running: git checkout autosynth
Switched to branch 'autosynth'
2020-06-20 01:34:19,076 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 367, in synthesize_loop
    synthesize_inner_loop(fork, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 411, in synthesize_inner_loop
    synthesizer, len(toolbox.versions) - 1
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 266, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 649, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 506, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 629, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 375, in synthesize_loop
    synthesize_loop_single_pr(toolbox, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 401, in synthesize_loop_single_pr
    synthesize_inner_loop(toolbox, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 411, in synthesize_inner_loop
    synthesizer, len(toolbox.versions) - 1
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 266, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/eeae4746-3508-43c9-a72e-6d6d9b4f6782/targets/github%2Fsynthtool;config=default/tests;query=google-http-java-client;failed=false).

Autosynth is still having trouble generating google-http-java-client. :sob:

Here's the output from running `synth.py`:

```
mpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1357, in __next__
    return self._next()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1125, in generate
    yield self.environment.handle_exception()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/home/kbuilder/.cache/synthtool/synthtool/synthtool/gcp/templates/java_library/.github/workflows/ci.yaml", line 12, in top-level template code
    java: [{% if metadata['min_java_version'] <= 7 %}7, {% endif %}8, 11]
jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'min_java_version'
2020-06-20 14:34:24,005 autosynth [ERROR] > Synthesis failed
2020-06-20 14:34:24,005 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at a82b980 chore(deps): update dependency org.apache.maven.plugins:maven-project-info-reports-plugin to v3.1.0 (#1050)
2020-06-20 14:34:24,013 autosynth [DEBUG] > Running: git checkout autosynth-synthtool
Switched to branch 'autosynth-synthtool'
2020-06-20 14:34:24,020 autosynth [ERROR] > Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.
2020-06-20 14:34:24,159 autosynth [DEBUG] > Running: git checkout a82b9808f112c240c213726356e610e3a9fb4e61
Note: checking out 'a82b9808f112c240c213726356e610e3a9fb4e61'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at a82b980 chore(deps): update dependency org.apache.maven.plugins:maven-project-info-reports-plugin to v3.1.0 (#1050)
2020-06-20 14:34:24,166 autosynth [DEBUG] > Running: git checkout 4f2c9f752a94042472fc03c5bd9e06e89817d2bd
HEAD is now at 4f2c9f7 ci(java): run dependency test on Java 8 and 11 (#639)
2020-06-20 14:34:24,173 autosynth [DEBUG] > Running: git branch -f autosynth-74
2020-06-20 14:34:24,176 autosynth [DEBUG] > Running: git checkout autosynth-74
Switched to branch 'autosynth-74'
2020-06-20 14:34:24,183 autosynth [INFO] > Running synthtool
2020-06-20 14:34:24,183 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']
2020-06-20 14:34:24,183 autosynth [DEBUG] > log_file_path: /tmpfs/src/github/synthtool/logs/googleapis/google-http-java-client/74/sponge_log.log
2020-06-20 14:34:24,185 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata synth.metadata synth.py --
2020-06-20 14:34:24,394 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/google-http-java-client/synth.py.
On branch autosynth-74
nothing to commit, working tree clean
2020-06-20 14:34:24,530 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
DEBUG:synthtool:Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
.github/CODEOWNERS
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/release-please.yml
.github/trusted-contribution.yml
.github/workflows/ci.yaml
2020-06-20 14:34:24,553 synthtool [DEBUG] > Wrote metadata to synth.metadata.
DEBUG:synthtool:Wrote metadata to synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/google-http-java-client/synth.py", line 21, in <module>
    templates = common_templates.java_library()
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 95, in java_library
    return self._generic_library("java_library", **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 57, in _generic_library
    result = t.render(**kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 83, in render
    _render_to_path(self.env, template_name, self.dir / subdir, kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 53, in _render_to_path
    output.dump(fh)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1313, in dump
    fp.writelines(iterable)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1357, in __next__
    return self._next()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1125, in generate
    yield self.environment.handle_exception()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/home/kbuilder/.cache/synthtool/synthtool/synthtool/gcp/templates/java_library/.github/workflows/ci.yaml", line 12, in top-level template code
    java: [{% if metadata['min_java_version'] <= 7 %}7, {% endif %}8, 11]
jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'min_java_version'
2020-06-20 14:34:24,595 autosynth [ERROR] > Synthesis failed
2020-06-20 14:34:24,596 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at a82b980 chore(deps): update dependency org.apache.maven.plugins:maven-project-info-reports-plugin to v3.1.0 (#1050)
2020-06-20 14:34:24,603 autosynth [DEBUG] > Running: git checkout autosynth
Switched to branch 'autosynth'
2020-06-20 14:34:24,609 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 367, in synthesize_loop
    synthesize_inner_loop(fork, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 411, in synthesize_inner_loop
    synthesizer, len(toolbox.versions) - 1
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 266, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 649, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 506, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 629, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 375, in synthesize_loop
    synthesize_loop_single_pr(toolbox, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 401, in synthesize_loop_single_pr
    synthesize_inner_loop(toolbox, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 411, in synthesize_inner_loop
    synthesizer, len(toolbox.versions) - 1
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 266, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/6d62f267-678f-42e1-b268-22936d8e9d75/targets/github%2Fsynthtool;config=default/tests;query=google-http-java-client;failed=false).

Autosynth is still having trouble generating google-http-java-client. :sob:

Here's the output from running `synth.py`:

```
mpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1357, in __next__
    return self._next()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1125, in generate
    yield self.environment.handle_exception()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/home/kbuilder/.cache/synthtool/synthtool/synthtool/gcp/templates/java_library/.github/workflows/ci.yaml", line 12, in top-level template code
    java: [{% if metadata['min_java_version'] <= 7 %}7, {% endif %}8, 11]
jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'min_java_version'
2020-06-21 01:34:23,268 autosynth [ERROR] > Synthesis failed
2020-06-21 01:34:23,268 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at a82b980 chore(deps): update dependency org.apache.maven.plugins:maven-project-info-reports-plugin to v3.1.0 (#1050)
2020-06-21 01:34:23,276 autosynth [DEBUG] > Running: git checkout autosynth-synthtool
Switched to branch 'autosynth-synthtool'
2020-06-21 01:34:23,283 autosynth [ERROR] > Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.
2020-06-21 01:34:23,419 autosynth [DEBUG] > Running: git checkout a82b9808f112c240c213726356e610e3a9fb4e61
Note: checking out 'a82b9808f112c240c213726356e610e3a9fb4e61'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at a82b980 chore(deps): update dependency org.apache.maven.plugins:maven-project-info-reports-plugin to v3.1.0 (#1050)
2020-06-21 01:34:23,426 autosynth [DEBUG] > Running: git checkout 4f2c9f752a94042472fc03c5bd9e06e89817d2bd
HEAD is now at 4f2c9f7 ci(java): run dependency test on Java 8 and 11 (#639)
2020-06-21 01:34:23,432 autosynth [DEBUG] > Running: git branch -f autosynth-74
2020-06-21 01:34:23,436 autosynth [DEBUG] > Running: git checkout autosynth-74
Switched to branch 'autosynth-74'
2020-06-21 01:34:23,443 autosynth [INFO] > Running synthtool
2020-06-21 01:34:23,443 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']
2020-06-21 01:34:23,443 autosynth [DEBUG] > log_file_path: /tmpfs/src/github/synthtool/logs/googleapis/google-http-java-client/74/sponge_log.log
2020-06-21 01:34:23,446 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata synth.metadata synth.py --
2020-06-21 01:34:23,661 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/google-http-java-client/synth.py.
On branch autosynth-74
nothing to commit, working tree clean
2020-06-21 01:34:23,800 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
DEBUG:synthtool:Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
.github/CODEOWNERS
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/release-please.yml
.github/trusted-contribution.yml
.github/workflows/ci.yaml
2020-06-21 01:34:23,824 synthtool [DEBUG] > Wrote metadata to synth.metadata.
DEBUG:synthtool:Wrote metadata to synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/google-http-java-client/synth.py", line 21, in <module>
    templates = common_templates.java_library()
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 95, in java_library
    return self._generic_library("java_library", **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 57, in _generic_library
    result = t.render(**kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 83, in render
    _render_to_path(self.env, template_name, self.dir / subdir, kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 53, in _render_to_path
    output.dump(fh)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1313, in dump
    fp.writelines(iterable)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1357, in __next__
    return self._next()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1125, in generate
    yield self.environment.handle_exception()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/home/kbuilder/.cache/synthtool/synthtool/synthtool/gcp/templates/java_library/.github/workflows/ci.yaml", line 12, in top-level template code
    java: [{% if metadata['min_java_version'] <= 7 %}7, {% endif %}8, 11]
jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'min_java_version'
2020-06-21 01:34:23,872 autosynth [ERROR] > Synthesis failed
2020-06-21 01:34:23,872 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at a82b980 chore(deps): update dependency org.apache.maven.plugins:maven-project-info-reports-plugin to v3.1.0 (#1050)
2020-06-21 01:34:23,879 autosynth [DEBUG] > Running: git checkout autosynth
Switched to branch 'autosynth'
2020-06-21 01:34:23,886 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 367, in synthesize_loop
    synthesize_inner_loop(fork, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 411, in synthesize_inner_loop
    synthesizer, len(toolbox.versions) - 1
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 266, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 649, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 506, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 629, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 375, in synthesize_loop
    synthesize_loop_single_pr(toolbox, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 401, in synthesize_loop_single_pr
    synthesize_inner_loop(toolbox, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 411, in synthesize_inner_loop
    synthesizer, len(toolbox.versions) - 1
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 266, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/a6999603-c2cf-4e41-b6d2-20fb36f13bc7/targets/github%2Fsynthtool;config=default/tests;query=google-http-java-client;failed=false).

Autosynth is still having trouble generating google-http-java-client. :sob:

Here's the output from running `synth.py`:

```
mpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1357, in __next__
    return self._next()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1125, in generate
    yield self.environment.handle_exception()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/home/kbuilder/.cache/synthtool/synthtool/synthtool/gcp/templates/java_library/.github/workflows/ci.yaml", line 12, in top-level template code
    java: [{% if metadata['min_java_version'] <= 7 %}7, {% endif %}8, 11]
jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'min_java_version'
2020-06-21 14:34:19,638 autosynth [ERROR] > Synthesis failed
2020-06-21 14:34:19,639 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at a82b980 chore(deps): update dependency org.apache.maven.plugins:maven-project-info-reports-plugin to v3.1.0 (#1050)
2020-06-21 14:34:19,645 autosynth [DEBUG] > Running: git checkout autosynth-synthtool
Switched to branch 'autosynth-synthtool'
2020-06-21 14:34:19,651 autosynth [ERROR] > Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.
2020-06-21 14:34:19,778 autosynth [DEBUG] > Running: git checkout a82b9808f112c240c213726356e610e3a9fb4e61
Note: checking out 'a82b9808f112c240c213726356e610e3a9fb4e61'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at a82b980 chore(deps): update dependency org.apache.maven.plugins:maven-project-info-reports-plugin to v3.1.0 (#1050)
2020-06-21 14:34:19,785 autosynth [DEBUG] > Running: git checkout 4f2c9f752a94042472fc03c5bd9e06e89817d2bd
HEAD is now at 4f2c9f7 ci(java): run dependency test on Java 8 and 11 (#639)
2020-06-21 14:34:19,790 autosynth [DEBUG] > Running: git branch -f autosynth-74
2020-06-21 14:34:19,793 autosynth [DEBUG] > Running: git checkout autosynth-74
Switched to branch 'autosynth-74'
2020-06-21 14:34:19,799 autosynth [INFO] > Running synthtool
2020-06-21 14:34:19,799 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']
2020-06-21 14:34:19,799 autosynth [DEBUG] > log_file_path: /tmpfs/src/github/synthtool/logs/googleapis/google-http-java-client/74/sponge_log.log
2020-06-21 14:34:19,801 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata synth.metadata synth.py --
2020-06-21 14:34:20,018 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/google-http-java-client/synth.py.
On branch autosynth-74
nothing to commit, working tree clean
2020-06-21 14:34:20,158 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
DEBUG:synthtool:Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
.github/CODEOWNERS
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/release-please.yml
.github/trusted-contribution.yml
.github/workflows/ci.yaml
2020-06-21 14:34:20,180 synthtool [DEBUG] > Wrote metadata to synth.metadata.
DEBUG:synthtool:Wrote metadata to synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/google-http-java-client/synth.py", line 21, in <module>
    templates = common_templates.java_library()
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 95, in java_library
    return self._generic_library("java_library", **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 57, in _generic_library
    result = t.render(**kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 83, in render
    _render_to_path(self.env, template_name, self.dir / subdir, kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 53, in _render_to_path
    output.dump(fh)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1313, in dump
    fp.writelines(iterable)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1357, in __next__
    return self._next()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1125, in generate
    yield self.environment.handle_exception()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/home/kbuilder/.cache/synthtool/synthtool/synthtool/gcp/templates/java_library/.github/workflows/ci.yaml", line 12, in top-level template code
    java: [{% if metadata['min_java_version'] <= 7 %}7, {% endif %}8, 11]
jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'min_java_version'
2020-06-21 14:34:20,215 autosynth [ERROR] > Synthesis failed
2020-06-21 14:34:20,216 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at a82b980 chore(deps): update dependency org.apache.maven.plugins:maven-project-info-reports-plugin to v3.1.0 (#1050)
2020-06-21 14:34:20,222 autosynth [DEBUG] > Running: git checkout autosynth
Switched to branch 'autosynth'
2020-06-21 14:34:20,228 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 367, in synthesize_loop
    synthesize_inner_loop(fork, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 411, in synthesize_inner_loop
    synthesizer, len(toolbox.versions) - 1
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 266, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 649, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 506, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 629, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 375, in synthesize_loop
    synthesize_loop_single_pr(toolbox, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 401, in synthesize_loop_single_pr
    synthesize_inner_loop(toolbox, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 411, in synthesize_inner_loop
    synthesizer, len(toolbox.versions) - 1
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 266, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/a27f6149-a4c2-400f-86da-f3983d8cb77a/targets/github%2Fsynthtool;config=default/tests;query=google-http-java-client;failed=false).

Autosynth is still having trouble generating google-http-java-client. :sob:

Here's the output from running `synth.py`:

```
mpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1357, in __next__
    return self._next()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1125, in generate
    yield self.environment.handle_exception()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/home/kbuilder/.cache/synthtool/synthtool/synthtool/gcp/templates/java_library/.github/workflows/ci.yaml", line 12, in top-level template code
    java: [{% if metadata['min_java_version'] <= 7 %}7, {% endif %}8, 11]
jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'min_java_version'
2020-06-22 01:34:23,317 autosynth [ERROR] > Synthesis failed
2020-06-22 01:34:23,318 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at a82b980 chore(deps): update dependency org.apache.maven.plugins:maven-project-info-reports-plugin to v3.1.0 (#1050)
2020-06-22 01:34:23,325 autosynth [DEBUG] > Running: git checkout autosynth-synthtool
Switched to branch 'autosynth-synthtool'
2020-06-22 01:34:23,332 autosynth [ERROR] > Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.
2020-06-22 01:34:23,476 autosynth [DEBUG] > Running: git checkout a82b9808f112c240c213726356e610e3a9fb4e61
Note: checking out 'a82b9808f112c240c213726356e610e3a9fb4e61'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at a82b980 chore(deps): update dependency org.apache.maven.plugins:maven-project-info-reports-plugin to v3.1.0 (#1050)
2020-06-22 01:34:23,483 autosynth [DEBUG] > Running: git checkout 4f2c9f752a94042472fc03c5bd9e06e89817d2bd
HEAD is now at 4f2c9f7 ci(java): run dependency test on Java 8 and 11 (#639)
2020-06-22 01:34:23,494 autosynth [DEBUG] > Running: git branch -f autosynth-74
2020-06-22 01:34:23,497 autosynth [DEBUG] > Running: git checkout autosynth-74
Switched to branch 'autosynth-74'
2020-06-22 01:34:23,503 autosynth [INFO] > Running synthtool
2020-06-22 01:34:23,504 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']
2020-06-22 01:34:23,504 autosynth [DEBUG] > log_file_path: /tmpfs/src/github/synthtool/logs/googleapis/google-http-java-client/74/sponge_log.log
2020-06-22 01:34:23,505 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata synth.metadata synth.py --
2020-06-22 01:34:23,703 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/google-http-java-client/synth.py.
On branch autosynth-74
nothing to commit, working tree clean
2020-06-22 01:34:23,830 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
DEBUG:synthtool:Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
.github/CODEOWNERS
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/release-please.yml
.github/trusted-contribution.yml
.github/workflows/ci.yaml
2020-06-22 01:34:23,851 synthtool [DEBUG] > Wrote metadata to synth.metadata.
DEBUG:synthtool:Wrote metadata to synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/google-http-java-client/synth.py", line 21, in <module>
    templates = common_templates.java_library()
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 95, in java_library
    return self._generic_library("java_library", **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 57, in _generic_library
    result = t.render(**kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 83, in render
    _render_to_path(self.env, template_name, self.dir / subdir, kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 53, in _render_to_path
    output.dump(fh)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1313, in dump
    fp.writelines(iterable)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1357, in __next__
    return self._next()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1125, in generate
    yield self.environment.handle_exception()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/home/kbuilder/.cache/synthtool/synthtool/synthtool/gcp/templates/java_library/.github/workflows/ci.yaml", line 12, in top-level template code
    java: [{% if metadata['min_java_version'] <= 7 %}7, {% endif %}8, 11]
jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'min_java_version'
2020-06-22 01:34:23,890 autosynth [ERROR] > Synthesis failed
2020-06-22 01:34:23,890 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at a82b980 chore(deps): update dependency org.apache.maven.plugins:maven-project-info-reports-plugin to v3.1.0 (#1050)
2020-06-22 01:34:23,897 autosynth [DEBUG] > Running: git checkout autosynth
Switched to branch 'autosynth'
2020-06-22 01:34:23,903 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 367, in synthesize_loop
    synthesize_inner_loop(fork, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 411, in synthesize_inner_loop
    synthesizer, len(toolbox.versions) - 1
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 266, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 649, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 506, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 629, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 375, in synthesize_loop
    synthesize_loop_single_pr(toolbox, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 401, in synthesize_loop_single_pr
    synthesize_inner_loop(toolbox, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 411, in synthesize_inner_loop
    synthesizer, len(toolbox.versions) - 1
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 266, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/3fff4fa1-4c3b-4e98-bd4f-c465ef9d56d1/targets/github%2Fsynthtool;config=default/tests;query=google-http-java-client;failed=false).

Autosynth is still having trouble generating google-http-java-client. :sob:

Here's the output from running `synth.py`:

```
mpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1357, in __next__
    return self._next()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1125, in generate
    yield self.environment.handle_exception()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/home/kbuilder/.cache/synthtool/synthtool/synthtool/gcp/templates/java_library/.github/workflows/ci.yaml", line 12, in top-level template code
    java: [{% if metadata['min_java_version'] <= 7 %}7, {% endif %}8, 11]
jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'min_java_version'
2020-06-22 14:34:19,182 autosynth [ERROR] > Synthesis failed
2020-06-22 14:34:19,182 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at a82b980 chore(deps): update dependency org.apache.maven.plugins:maven-project-info-reports-plugin to v3.1.0 (#1050)
2020-06-22 14:34:19,189 autosynth [DEBUG] > Running: git checkout autosynth-synthtool
Switched to branch 'autosynth-synthtool'
2020-06-22 14:34:19,195 autosynth [ERROR] > Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.
2020-06-22 14:34:19,326 autosynth [DEBUG] > Running: git checkout a82b9808f112c240c213726356e610e3a9fb4e61
Note: checking out 'a82b9808f112c240c213726356e610e3a9fb4e61'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at a82b980 chore(deps): update dependency org.apache.maven.plugins:maven-project-info-reports-plugin to v3.1.0 (#1050)
2020-06-22 14:34:19,334 autosynth [DEBUG] > Running: git checkout 4f2c9f752a94042472fc03c5bd9e06e89817d2bd
HEAD is now at 4f2c9f7 ci(java): run dependency test on Java 8 and 11 (#639)
2020-06-22 14:34:19,339 autosynth [DEBUG] > Running: git branch -f autosynth-74
2020-06-22 14:34:19,342 autosynth [DEBUG] > Running: git checkout autosynth-74
Switched to branch 'autosynth-74'
2020-06-22 14:34:19,348 autosynth [INFO] > Running synthtool
2020-06-22 14:34:19,348 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']
2020-06-22 14:34:19,349 autosynth [DEBUG] > log_file_path: /tmpfs/src/github/synthtool/logs/googleapis/google-http-java-client/74/sponge_log.log
2020-06-22 14:34:19,350 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata synth.metadata synth.py --
2020-06-22 14:34:19,569 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/google-http-java-client/synth.py.
On branch autosynth-74
nothing to commit, working tree clean
2020-06-22 14:34:19,707 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
DEBUG:synthtool:Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
.github/CODEOWNERS
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/release-please.yml
.github/trusted-contribution.yml
.github/workflows/ci.yaml
2020-06-22 14:34:19,729 synthtool [DEBUG] > Wrote metadata to synth.metadata.
DEBUG:synthtool:Wrote metadata to synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/google-http-java-client/synth.py", line 21, in <module>
    templates = common_templates.java_library()
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 95, in java_library
    return self._generic_library("java_library", **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 57, in _generic_library
    result = t.render(**kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 83, in render
    _render_to_path(self.env, template_name, self.dir / subdir, kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 53, in _render_to_path
    output.dump(fh)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1313, in dump
    fp.writelines(iterable)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1357, in __next__
    return self._next()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1125, in generate
    yield self.environment.handle_exception()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/home/kbuilder/.cache/synthtool/synthtool/synthtool/gcp/templates/java_library/.github/workflows/ci.yaml", line 12, in top-level template code
    java: [{% if metadata['min_java_version'] <= 7 %}7, {% endif %}8, 11]
jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'min_java_version'
2020-06-22 14:34:19,765 autosynth [ERROR] > Synthesis failed
2020-06-22 14:34:19,765 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at a82b980 chore(deps): update dependency org.apache.maven.plugins:maven-project-info-reports-plugin to v3.1.0 (#1050)
2020-06-22 14:34:19,773 autosynth [DEBUG] > Running: git checkout autosynth
Switched to branch 'autosynth'
2020-06-22 14:34:19,778 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 367, in synthesize_loop
    synthesize_inner_loop(fork, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 411, in synthesize_inner_loop
    synthesizer, len(toolbox.versions) - 1
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 266, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 649, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 506, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 629, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 375, in synthesize_loop
    synthesize_loop_single_pr(toolbox, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 401, in synthesize_loop_single_pr
    synthesize_inner_loop(toolbox, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 411, in synthesize_inner_loop
    synthesizer, len(toolbox.versions) - 1
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 266, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/a3e09c0b-006d-4d98-8e96-744af862718e/targets/github%2Fsynthtool;config=default/tests;query=google-http-java-client;failed=false).

Autosynth is still having trouble generating google-http-java-client. :sob:

Here's the output from running `synth.py`:

```
ithub/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1357, in __next__
    return self._next()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1125, in generate
    yield self.environment.handle_exception()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/home/kbuilder/.cache/synthtool/synthtool/synthtool/gcp/templates/java_library/.github/workflows/ci.yaml", line 12, in top-level template code
    java: [{% if metadata['min_java_version'] <= 7 %}7, {% endif %}8, 11]
jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'min_java_version'
2020-06-23 01:34:17,679 autosynth [ERROR] > Synthesis failed
2020-06-23 01:34:17,679 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at a82b980 chore(deps): update dependency org.apache.maven.plugins:maven-project-info-reports-plugin to v3.1.0 (#1050)
2020-06-23 01:34:17,685 autosynth [DEBUG] > Running: git checkout autosynth-synthtool
Switched to branch 'autosynth-synthtool'
2020-06-23 01:34:17,691 autosynth [ERROR] > Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.
2020-06-23 01:34:17,832 autosynth [DEBUG] > Running: git checkout a82b9808f112c240c213726356e610e3a9fb4e61
Note: checking out 'a82b9808f112c240c213726356e610e3a9fb4e61'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at a82b980 chore(deps): update dependency org.apache.maven.plugins:maven-project-info-reports-plugin to v3.1.0 (#1050)
2020-06-23 01:34:17,838 autosynth [DEBUG] > Running: git checkout db4f15fedc76e04018e6e43ec403e804e1142158
HEAD is now at db4f15f feat(python): add defaults for microgenerator libraries (#642)
2020-06-23 01:34:17,843 autosynth [DEBUG] > Running: git branch -f autosynth-75
2020-06-23 01:34:17,846 autosynth [DEBUG] > Running: git checkout autosynth-75
Switched to branch 'autosynth-75'
2020-06-23 01:34:17,851 autosynth [INFO] > Running synthtool
2020-06-23 01:34:17,852 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']
2020-06-23 01:34:17,852 autosynth [DEBUG] > log_file_path: /tmpfs/src/github/synthtool/logs/googleapis/google-http-java-client/75/sponge_log.log
2020-06-23 01:34:17,853 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata synth.metadata synth.py --
2020-06-23 01:34:18,063 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/google-http-java-client/synth.py.
On branch autosynth-75
nothing to commit, working tree clean
2020-06-23 01:34:18,196 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
DEBUG:synthtool:Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
.github/CODEOWNERS
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/release-please.yml
.github/trusted-contribution.yml
.github/workflows/ci.yaml
2020-06-23 01:34:18,218 synthtool [DEBUG] > Wrote metadata to synth.metadata.
DEBUG:synthtool:Wrote metadata to synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/google-http-java-client/synth.py", line 21, in <module>
    templates = common_templates.java_library()
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 104, in java_library
    return self._generic_library("java_library", **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 57, in _generic_library
    result = t.render(**kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 83, in render
    _render_to_path(self.env, template_name, self.dir / subdir, kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 53, in _render_to_path
    output.dump(fh)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1313, in dump
    fp.writelines(iterable)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1357, in __next__
    return self._next()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1125, in generate
    yield self.environment.handle_exception()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/home/kbuilder/.cache/synthtool/synthtool/synthtool/gcp/templates/java_library/.github/workflows/ci.yaml", line 12, in top-level template code
    java: [{% if metadata['min_java_version'] <= 7 %}7, {% endif %}8, 11]
jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'min_java_version'
2020-06-23 01:34:18,253 autosynth [ERROR] > Synthesis failed
2020-06-23 01:34:18,253 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at a82b980 chore(deps): update dependency org.apache.maven.plugins:maven-project-info-reports-plugin to v3.1.0 (#1050)
2020-06-23 01:34:18,260 autosynth [DEBUG] > Running: git checkout autosynth
Switched to branch 'autosynth'
2020-06-23 01:34:18,265 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 367, in synthesize_loop
    synthesize_inner_loop(fork, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 411, in synthesize_inner_loop
    synthesizer, len(toolbox.versions) - 1
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 266, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 649, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 506, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 629, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 375, in synthesize_loop
    synthesize_loop_single_pr(toolbox, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 401, in synthesize_loop_single_pr
    synthesize_inner_loop(toolbox, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 411, in synthesize_inner_loop
    synthesizer, len(toolbox.versions) - 1
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 266, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/26ec1231-fc23-4217-a1d8-d3014038092b/targets/github%2Fsynthtool;config=default/tests;query=google-http-java-client;failed=false).

Autosynth is still having trouble generating google-http-java-client. :sob:

Here's the output from running `synth.py`:

```
ble)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1357, in __next__
    return self._next()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1125, in generate
    yield self.environment.handle_exception()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/home/kbuilder/.cache/synthtool/synthtool/synthtool/gcp/templates/java_library/.github/workflows/ci.yaml", line 12, in top-level template code
    java: [{% if metadata['min_java_version'] <= 7 %}7, {% endif %}8, 11]
jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'min_java_version'
2020-06-23 14:36:35,350 autosynth [ERROR] > Synthesis failed
2020-06-23 14:36:35,350 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at a82b980 chore(deps): update dependency org.apache.maven.plugins:maven-project-info-reports-plugin to v3.1.0 (#1050)
2020-06-23 14:36:35,357 autosynth [DEBUG] > Running: git checkout autosynth-synthtool
Switched to branch 'autosynth-synthtool'
2020-06-23 14:36:35,363 autosynth [ERROR] > Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.
2020-06-23 14:36:35,532 autosynth [DEBUG] > Running: git checkout a82b9808f112c240c213726356e610e3a9fb4e61
Note: checking out 'a82b9808f112c240c213726356e610e3a9fb4e61'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at a82b980 chore(deps): update dependency org.apache.maven.plugins:maven-project-info-reports-plugin to v3.1.0 (#1050)
2020-06-23 14:36:35,539 autosynth [DEBUG] > Running: git checkout 2a29860e484695a07aa8670f9e750bfeeb308a43
HEAD is now at 2a29860 build: install multiple Python versions for autosynth runs (#641)
2020-06-23 14:36:35,549 autosynth [DEBUG] > Running: git branch -f autosynth-78
2020-06-23 14:36:35,552 autosynth [DEBUG] > Running: git checkout autosynth-78
Switched to branch 'autosynth-78'
2020-06-23 14:36:35,558 autosynth [INFO] > Running synthtool
2020-06-23 14:36:35,558 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']
2020-06-23 14:36:35,558 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/google-http-java-client/78/sponge_log.log
2020-06-23 14:36:35,560 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata synth.metadata synth.py --
2020-06-23 14:36:35,755 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/google-http-java-client/synth.py.
On branch autosynth-78
nothing to commit, working tree clean
2020-06-23 14:36:35,885 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
DEBUG:synthtool:Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
.github/CODEOWNERS
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/release-please.yml
.github/trusted-contribution.yml
.github/workflows/ci.yaml
2020-06-23 14:36:35,906 synthtool [DEBUG] > Wrote metadata to synth.metadata.
DEBUG:synthtool:Wrote metadata to synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/google-http-java-client/synth.py", line 21, in <module>
    templates = common_templates.java_library()
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 104, in java_library
    return self._generic_library("java_library", **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 57, in _generic_library
    result = t.render(**kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 83, in render
    _render_to_path(self.env, template_name, self.dir / subdir, kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 53, in _render_to_path
    output.dump(fh)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1313, in dump
    fp.writelines(iterable)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1357, in __next__
    return self._next()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1125, in generate
    yield self.environment.handle_exception()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/home/kbuilder/.cache/synthtool/synthtool/synthtool/gcp/templates/java_library/.github/workflows/ci.yaml", line 12, in top-level template code
    java: [{% if metadata['min_java_version'] <= 7 %}7, {% endif %}8, 11]
jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'min_java_version'
2020-06-23 14:36:35,944 autosynth [ERROR] > Synthesis failed
2020-06-23 14:36:35,944 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at a82b980 chore(deps): update dependency org.apache.maven.plugins:maven-project-info-reports-plugin to v3.1.0 (#1050)
2020-06-23 14:36:35,951 autosynth [DEBUG] > Running: git checkout autosynth
Switched to branch 'autosynth'
2020-06-23 14:36:35,957 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 367, in synthesize_loop
    synthesize_inner_loop(fork, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 411, in synthesize_inner_loop
    synthesizer, len(toolbox.versions) - 1
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 266, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 657, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 506, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 637, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 375, in synthesize_loop
    synthesize_loop_single_pr(toolbox, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 401, in synthesize_loop_single_pr
    synthesize_inner_loop(toolbox, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 411, in synthesize_inner_loop
    synthesizer, len(toolbox.versions) - 1
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 266, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/9c18038c-d1ba-45ab-9ded-a79e3b7a4d1b/targets/github%2Fsynthtool;config=default/tests;query=google-http-java-client;failed=false).

Autosynth is still having trouble generating google-http-java-client. :sob:

Here's the output from running `synth.py`:

```
ble)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1357, in __next__
    return self._next()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1125, in generate
    yield self.environment.handle_exception()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/home/kbuilder/.cache/synthtool/synthtool/synthtool/gcp/templates/java_library/.github/workflows/ci.yaml", line 12, in top-level template code
    java: [{% if metadata['min_java_version'] <= 7 %}7, {% endif %}8, 11]
jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'min_java_version'
2020-06-24 01:34:26,948 autosynth [ERROR] > Synthesis failed
2020-06-24 01:34:26,949 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at a82b980 chore(deps): update dependency org.apache.maven.plugins:maven-project-info-reports-plugin to v3.1.0 (#1050)
2020-06-24 01:34:26,955 autosynth [DEBUG] > Running: git checkout autosynth-synthtool
Switched to branch 'autosynth-synthtool'
2020-06-24 01:34:26,962 autosynth [ERROR] > Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.
2020-06-24 01:34:27,096 autosynth [DEBUG] > Running: git checkout a82b9808f112c240c213726356e610e3a9fb4e61
Note: checking out 'a82b9808f112c240c213726356e610e3a9fb4e61'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at a82b980 chore(deps): update dependency org.apache.maven.plugins:maven-project-info-reports-plugin to v3.1.0 (#1050)
2020-06-24 01:34:27,103 autosynth [DEBUG] > Running: git checkout 2a29860e484695a07aa8670f9e750bfeeb308a43
HEAD is now at 2a29860 build: install multiple Python versions for autosynth runs (#641)
2020-06-24 01:34:27,109 autosynth [DEBUG] > Running: git branch -f autosynth-78
2020-06-24 01:34:27,112 autosynth [DEBUG] > Running: git checkout autosynth-78
Switched to branch 'autosynth-78'
2020-06-24 01:34:27,118 autosynth [INFO] > Running synthtool
2020-06-24 01:34:27,119 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']
2020-06-24 01:34:27,119 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/google-http-java-client/78/sponge_log.log
2020-06-24 01:34:27,121 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata synth.metadata synth.py --
2020-06-24 01:34:27,346 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/google-http-java-client/synth.py.
On branch autosynth-78
nothing to commit, working tree clean
2020-06-24 01:34:27,491 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
DEBUG:synthtool:Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
.github/CODEOWNERS
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/release-please.yml
.github/trusted-contribution.yml
.github/workflows/ci.yaml
2020-06-24 01:34:27,515 synthtool [DEBUG] > Wrote metadata to synth.metadata.
DEBUG:synthtool:Wrote metadata to synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/google-http-java-client/synth.py", line 21, in <module>
    templates = common_templates.java_library()
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 104, in java_library
    return self._generic_library("java_library", **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 57, in _generic_library
    result = t.render(**kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 83, in render
    _render_to_path(self.env, template_name, self.dir / subdir, kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 53, in _render_to_path
    output.dump(fh)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1313, in dump
    fp.writelines(iterable)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1357, in __next__
    return self._next()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1125, in generate
    yield self.environment.handle_exception()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/home/kbuilder/.cache/synthtool/synthtool/synthtool/gcp/templates/java_library/.github/workflows/ci.yaml", line 12, in top-level template code
    java: [{% if metadata['min_java_version'] <= 7 %}7, {% endif %}8, 11]
jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'min_java_version'
2020-06-24 01:34:27,556 autosynth [ERROR] > Synthesis failed
2020-06-24 01:34:27,557 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at a82b980 chore(deps): update dependency org.apache.maven.plugins:maven-project-info-reports-plugin to v3.1.0 (#1050)
2020-06-24 01:34:27,563 autosynth [DEBUG] > Running: git checkout autosynth
Switched to branch 'autosynth'
2020-06-24 01:34:27,570 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 367, in synthesize_loop
    synthesize_inner_loop(fork, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 411, in synthesize_inner_loop
    synthesizer, len(toolbox.versions) - 1
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 266, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 657, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 506, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 637, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 375, in synthesize_loop
    synthesize_loop_single_pr(toolbox, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 401, in synthesize_loop_single_pr
    synthesize_inner_loop(toolbox, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 411, in synthesize_inner_loop
    synthesizer, len(toolbox.versions) - 1
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 266, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/24de3e05-3efb-415a-ac3d-99f7d576a03f/targets/github%2Fsynthtool;config=default/tests;query=google-http-java-client;failed=false).

Autosynth is still having trouble generating google-http-java-client. :sob:

Here's the output from running `synth.py`:

```
pfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1357, in __next__
    return self._next()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1125, in generate
    yield self.environment.handle_exception()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/home/kbuilder/.cache/synthtool/synthtool/synthtool/gcp/templates/java_library/.github/workflows/ci.yaml", line 12, in top-level template code
    java: [{% if metadata['min_java_version'] <= 7 %}7, {% endif %}8, 11]
jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'min_java_version'
2020-06-24 14:34:39,913 autosynth [ERROR] > Synthesis failed
2020-06-24 14:34:39,913 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at a82b980 chore(deps): update dependency org.apache.maven.plugins:maven-project-info-reports-plugin to v3.1.0 (#1050)
2020-06-24 14:34:39,920 autosynth [DEBUG] > Running: git checkout autosynth-synthtool
Switched to branch 'autosynth-synthtool'
2020-06-24 14:34:39,926 autosynth [ERROR] > Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.
2020-06-24 14:34:40,061 autosynth [DEBUG] > Running: git checkout a82b9808f112c240c213726356e610e3a9fb4e61
Note: checking out 'a82b9808f112c240c213726356e610e3a9fb4e61'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at a82b980 chore(deps): update dependency org.apache.maven.plugins:maven-project-info-reports-plugin to v3.1.0 (#1050)
2020-06-24 14:34:40,068 autosynth [DEBUG] > Running: git checkout ce68c0e70d36c93ffcde96e9908fb4d94aa4f2e4
HEAD is now at ce68c0e Revert "build: install multiple Python versions for autosynth runs (#641)" (#646)
2020-06-24 14:34:40,074 autosynth [DEBUG] > Running: git branch -f autosynth-79
2020-06-24 14:34:40,077 autosynth [DEBUG] > Running: git checkout autosynth-79
Switched to branch 'autosynth-79'
2020-06-24 14:34:40,083 autosynth [INFO] > Running synthtool
2020-06-24 14:34:40,083 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']
2020-06-24 14:34:40,084 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/google-http-java-client/79/sponge_log.log
2020-06-24 14:34:40,086 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata synth.metadata synth.py --
2020-06-24 14:34:40,304 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/google-http-java-client/synth.py.
On branch autosynth-79
nothing to commit, working tree clean
2020-06-24 14:34:40,442 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
DEBUG:synthtool:Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
.github/CODEOWNERS
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/release-please.yml
.github/trusted-contribution.yml
.github/workflows/ci.yaml
2020-06-24 14:34:40,464 synthtool [DEBUG] > Wrote metadata to synth.metadata.
DEBUG:synthtool:Wrote metadata to synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/google-http-java-client/synth.py", line 21, in <module>
    templates = common_templates.java_library()
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 104, in java_library
    return self._generic_library("java_library", **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 57, in _generic_library
    result = t.render(**kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 83, in render
    _render_to_path(self.env, template_name, self.dir / subdir, kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 53, in _render_to_path
    output.dump(fh)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1313, in dump
    fp.writelines(iterable)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1357, in __next__
    return self._next()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1125, in generate
    yield self.environment.handle_exception()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/home/kbuilder/.cache/synthtool/synthtool/synthtool/gcp/templates/java_library/.github/workflows/ci.yaml", line 12, in top-level template code
    java: [{% if metadata['min_java_version'] <= 7 %}7, {% endif %}8, 11]
jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'min_java_version'
2020-06-24 14:34:40,501 autosynth [ERROR] > Synthesis failed
2020-06-24 14:34:40,502 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at a82b980 chore(deps): update dependency org.apache.maven.plugins:maven-project-info-reports-plugin to v3.1.0 (#1050)
2020-06-24 14:34:40,508 autosynth [DEBUG] > Running: git checkout autosynth
Switched to branch 'autosynth'
2020-06-24 14:34:40,514 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 367, in synthesize_loop
    synthesize_inner_loop(fork, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 411, in synthesize_inner_loop
    synthesizer, len(toolbox.versions) - 1
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 266, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 657, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 506, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 637, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 375, in synthesize_loop
    synthesize_loop_single_pr(toolbox, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 401, in synthesize_loop_single_pr
    synthesize_inner_loop(toolbox, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 411, in synthesize_inner_loop
    synthesizer, len(toolbox.versions) - 1
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 266, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/3a5af719-ea67-4913-a54a-a0061a536eed/targets/github%2Fsynthtool;config=default/tests;query=google-http-java-client;failed=false).

Autosynth is still having trouble generating google-http-java-client. :sob:

Here's the output from running `synth.py`:

```
pfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1357, in __next__
    return self._next()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1125, in generate
    yield self.environment.handle_exception()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/home/kbuilder/.cache/synthtool/synthtool/synthtool/gcp/templates/java_library/.github/workflows/ci.yaml", line 12, in top-level template code
    java: [{% if metadata['min_java_version'] <= 7 %}7, {% endif %}8, 11]
jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'min_java_version'
2020-06-25 01:34:07,311 autosynth [ERROR] > Synthesis failed
2020-06-25 01:34:07,311 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at a82b980 chore(deps): update dependency org.apache.maven.plugins:maven-project-info-reports-plugin to v3.1.0 (#1050)
2020-06-25 01:34:07,318 autosynth [DEBUG] > Running: git checkout autosynth-synthtool
Switched to branch 'autosynth-synthtool'
2020-06-25 01:34:07,324 autosynth [ERROR] > Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.
2020-06-25 01:34:07,452 autosynth [DEBUG] > Running: git checkout a82b9808f112c240c213726356e610e3a9fb4e61
Note: checking out 'a82b9808f112c240c213726356e610e3a9fb4e61'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at a82b980 chore(deps): update dependency org.apache.maven.plugins:maven-project-info-reports-plugin to v3.1.0 (#1050)
2020-06-25 01:34:07,458 autosynth [DEBUG] > Running: git checkout ce68c0e70d36c93ffcde96e9908fb4d94aa4f2e4
HEAD is now at ce68c0e Revert "build: install multiple Python versions for autosynth runs (#641)" (#646)
2020-06-25 01:34:07,469 autosynth [DEBUG] > Running: git branch -f autosynth-79
2020-06-25 01:34:07,472 autosynth [DEBUG] > Running: git checkout autosynth-79
Switched to branch 'autosynth-79'
2020-06-25 01:34:07,478 autosynth [INFO] > Running synthtool
2020-06-25 01:34:07,478 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']
2020-06-25 01:34:07,478 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/google-http-java-client/79/sponge_log.log
2020-06-25 01:34:07,480 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata synth.metadata synth.py --
2020-06-25 01:34:07,698 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/google-http-java-client/synth.py.
On branch autosynth-79
nothing to commit, working tree clean
2020-06-25 01:34:07,834 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
DEBUG:synthtool:Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
.github/CODEOWNERS
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/release-please.yml
.github/trusted-contribution.yml
.github/workflows/ci.yaml
2020-06-25 01:34:07,857 synthtool [DEBUG] > Wrote metadata to synth.metadata.
DEBUG:synthtool:Wrote metadata to synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/google-http-java-client/synth.py", line 21, in <module>
    templates = common_templates.java_library()
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 104, in java_library
    return self._generic_library("java_library", **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 57, in _generic_library
    result = t.render(**kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 83, in render
    _render_to_path(self.env, template_name, self.dir / subdir, kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 53, in _render_to_path
    output.dump(fh)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1313, in dump
    fp.writelines(iterable)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1357, in __next__
    return self._next()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1125, in generate
    yield self.environment.handle_exception()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/home/kbuilder/.cache/synthtool/synthtool/synthtool/gcp/templates/java_library/.github/workflows/ci.yaml", line 12, in top-level template code
    java: [{% if metadata['min_java_version'] <= 7 %}7, {% endif %}8, 11]
jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'min_java_version'
2020-06-25 01:34:07,895 autosynth [ERROR] > Synthesis failed
2020-06-25 01:34:07,895 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at a82b980 chore(deps): update dependency org.apache.maven.plugins:maven-project-info-reports-plugin to v3.1.0 (#1050)
2020-06-25 01:34:07,902 autosynth [DEBUG] > Running: git checkout autosynth
Switched to branch 'autosynth'
2020-06-25 01:34:07,908 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 367, in synthesize_loop
    synthesize_inner_loop(fork, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 411, in synthesize_inner_loop
    synthesizer, len(toolbox.versions) - 1
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 266, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 657, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 506, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 637, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 375, in synthesize_loop
    synthesize_loop_single_pr(toolbox, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 401, in synthesize_loop_single_pr
    synthesize_inner_loop(toolbox, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 411, in synthesize_inner_loop
    synthesizer, len(toolbox.versions) - 1
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 266, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/ed269a85-39ea-4dd1-9c6f-2c80320cbb34/targets/github%2Fsynthtool;config=default/tests;query=google-http-java-client;failed=false).

Autosynth is still having trouble generating google-http-java-client. :sob:

Here's the output from running `synth.py`:

```
", line 53, in _render_to_path
    output.dump(fh)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1313, in dump
    fp.writelines(iterable)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1357, in __next__
    return self._next()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1125, in generate
    yield self.environment.handle_exception()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/home/kbuilder/.cache/synthtool/synthtool/synthtool/gcp/templates/java_library/.github/workflows/ci.yaml", line 12, in top-level template code
    java: [{% if metadata['min_java_version'] <= 7 %}7, {% endif %}8, 11]
jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'min_java_version'
2020-06-25 11:56:30,720 autosynth [ERROR] > Synthesis failed
2020-06-25 11:56:30,720 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at b1353f8 chore(docs): libraries-bom 8.0.0 (#1066)
2020-06-25 11:56:30,727 autosynth [DEBUG] > Running: git checkout autosynth-synthtool
Switched to branch 'autosynth-synthtool'
2020-06-25 11:56:30,733 autosynth [ERROR] > Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.
2020-06-25 11:56:30,872 autosynth [DEBUG] > Running: git checkout b1353f8cc7aaf22a30b31bd2ac72248d3a3a0a5d
Note: checking out 'b1353f8cc7aaf22a30b31bd2ac72248d3a3a0a5d'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at b1353f8 chore(docs): libraries-bom 8.0.0 (#1066)
2020-06-25 11:56:30,879 autosynth [DEBUG] > Running: git checkout ce68c0e70d36c93ffcde96e9908fb4d94aa4f2e4
HEAD is now at ce68c0e Revert "build: install multiple Python versions for autosynth runs (#641)" (#646)
2020-06-25 11:56:30,884 autosynth [DEBUG] > Running: git branch -f autosynth-79
2020-06-25 11:56:30,887 autosynth [DEBUG] > Running: git checkout autosynth-79
Switched to branch 'autosynth-79'
2020-06-25 11:56:30,893 autosynth [INFO] > Running synthtool
2020-06-25 11:56:30,893 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']
2020-06-25 11:56:30,893 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/google-http-java-client/79/sponge_log.log
2020-06-25 11:56:30,895 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata synth.metadata synth.py --
2020-06-25 11:56:31,111 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/google-http-java-client/synth.py.
On branch autosynth-79
nothing to commit, working tree clean
2020-06-25 11:56:31,252 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
DEBUG:synthtool:Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
.github/CODEOWNERS
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/release-please.yml
.github/trusted-contribution.yml
.github/workflows/ci.yaml
2020-06-25 11:56:31,275 synthtool [DEBUG] > Wrote metadata to synth.metadata.
DEBUG:synthtool:Wrote metadata to synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/google-http-java-client/synth.py", line 21, in <module>
    templates = common_templates.java_library()
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 104, in java_library
    return self._generic_library("java_library", **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 57, in _generic_library
    result = t.render(**kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 83, in render
    _render_to_path(self.env, template_name, self.dir / subdir, kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 53, in _render_to_path
    output.dump(fh)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1313, in dump
    fp.writelines(iterable)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1357, in __next__
    return self._next()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1125, in generate
    yield self.environment.handle_exception()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/home/kbuilder/.cache/synthtool/synthtool/synthtool/gcp/templates/java_library/.github/workflows/ci.yaml", line 12, in top-level template code
    java: [{% if metadata['min_java_version'] <= 7 %}7, {% endif %}8, 11]
jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'min_java_version'
2020-06-25 11:56:31,311 autosynth [ERROR] > Synthesis failed
2020-06-25 11:56:31,311 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at b1353f8 chore(docs): libraries-bom 8.0.0 (#1066)
2020-06-25 11:56:31,318 autosynth [DEBUG] > Running: git checkout autosynth
Switched to branch 'autosynth'
2020-06-25 11:56:31,324 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 367, in synthesize_loop
    synthesize_inner_loop(fork, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 411, in synthesize_inner_loop
    synthesizer, len(toolbox.versions) - 1
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 266, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 657, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 506, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 637, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 375, in synthesize_loop
    synthesize_loop_single_pr(toolbox, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 401, in synthesize_loop_single_pr
    synthesize_inner_loop(toolbox, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 411, in synthesize_inner_loop
    synthesizer, len(toolbox.versions) - 1
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 266, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/ab32bcf0-0b80-4de6-8e9a-805e404915b2/targets/github%2Fsynthtool;config=default/tests;query=google-http-java-client;failed=false).

Autosynth is still having trouble generating google-http-java-client. :sob:

Here's the output from running `synth.py`:

```
", line 53, in _render_to_path
    output.dump(fh)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1313, in dump
    fp.writelines(iterable)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1357, in __next__
    return self._next()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1125, in generate
    yield self.environment.handle_exception()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/home/kbuilder/.cache/synthtool/synthtool/synthtool/gcp/templates/java_library/.github/workflows/ci.yaml", line 12, in top-level template code
    java: [{% if metadata['min_java_version'] <= 7 %}7, {% endif %}8, 11]
jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'min_java_version'
2020-06-25 14:34:11,960 autosynth [ERROR] > Synthesis failed
2020-06-25 14:34:11,961 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at b1353f8 chore(docs): libraries-bom 8.0.0 (#1066)
2020-06-25 14:34:11,967 autosynth [DEBUG] > Running: git checkout autosynth-synthtool
Switched to branch 'autosynth-synthtool'
2020-06-25 14:34:11,973 autosynth [ERROR] > Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.
2020-06-25 14:34:12,111 autosynth [DEBUG] > Running: git checkout b1353f8cc7aaf22a30b31bd2ac72248d3a3a0a5d
Note: checking out 'b1353f8cc7aaf22a30b31bd2ac72248d3a3a0a5d'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at b1353f8 chore(docs): libraries-bom 8.0.0 (#1066)
2020-06-25 14:34:12,117 autosynth [DEBUG] > Running: git checkout ce68c0e70d36c93ffcde96e9908fb4d94aa4f2e4
HEAD is now at ce68c0e Revert "build: install multiple Python versions for autosynth runs (#641)" (#646)
2020-06-25 14:34:12,122 autosynth [DEBUG] > Running: git branch -f autosynth-79
2020-06-25 14:34:12,125 autosynth [DEBUG] > Running: git checkout autosynth-79
Switched to branch 'autosynth-79'
2020-06-25 14:34:12,131 autosynth [INFO] > Running synthtool
2020-06-25 14:34:12,131 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']
2020-06-25 14:34:12,132 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/google-http-java-client/79/sponge_log.log
2020-06-25 14:34:12,133 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata synth.metadata synth.py --
2020-06-25 14:34:12,353 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/google-http-java-client/synth.py.
On branch autosynth-79
nothing to commit, working tree clean
2020-06-25 14:34:12,491 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
DEBUG:synthtool:Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
.github/CODEOWNERS
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/release-please.yml
.github/trusted-contribution.yml
.github/workflows/ci.yaml
2020-06-25 14:34:12,513 synthtool [DEBUG] > Wrote metadata to synth.metadata.
DEBUG:synthtool:Wrote metadata to synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/google-http-java-client/synth.py", line 21, in <module>
    templates = common_templates.java_library()
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 104, in java_library
    return self._generic_library("java_library", **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 57, in _generic_library
    result = t.render(**kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 83, in render
    _render_to_path(self.env, template_name, self.dir / subdir, kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 53, in _render_to_path
    output.dump(fh)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1313, in dump
    fp.writelines(iterable)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1357, in __next__
    return self._next()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1125, in generate
    yield self.environment.handle_exception()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/home/kbuilder/.cache/synthtool/synthtool/synthtool/gcp/templates/java_library/.github/workflows/ci.yaml", line 12, in top-level template code
    java: [{% if metadata['min_java_version'] <= 7 %}7, {% endif %}8, 11]
jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'min_java_version'
2020-06-25 14:34:12,548 autosynth [ERROR] > Synthesis failed
2020-06-25 14:34:12,548 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at b1353f8 chore(docs): libraries-bom 8.0.0 (#1066)
2020-06-25 14:34:12,555 autosynth [DEBUG] > Running: git checkout autosynth
Switched to branch 'autosynth'
2020-06-25 14:34:12,561 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 367, in synthesize_loop
    synthesize_inner_loop(fork, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 411, in synthesize_inner_loop
    synthesizer, len(toolbox.versions) - 1
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 266, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 657, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 506, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 637, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 375, in synthesize_loop
    synthesize_loop_single_pr(toolbox, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 401, in synthesize_loop_single_pr
    synthesize_inner_loop(toolbox, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 411, in synthesize_inner_loop
    synthesizer, len(toolbox.versions) - 1
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 266, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/01099226-2dbf-4b09-9ae9-126cd75a1ddb/targets/github%2Fsynthtool;config=default/tests;query=google-http-java-client;failed=false).

Autosynth is still having trouble generating google-http-java-client. :sob:

Here's the output from running `synth.py`:

```
ool/sources/templates.py", line 53, in _render_to_path
    output.dump(fh)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1313, in dump
    fp.writelines(iterable)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1357, in __next__
    return self._next()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1125, in generate
    yield self.environment.handle_exception()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/home/kbuilder/.cache/synthtool/synthtool/synthtool/gcp/templates/java_library/.github/workflows/ci.yaml", line 12, in top-level template code
    java: [{% if metadata['min_java_version'] <= 7 %}7, {% endif %}8, 11]
jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'min_java_version'
2020-06-26 01:34:16,377 autosynth [ERROR] > Synthesis failed
2020-06-26 01:34:16,377 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at b1353f8 chore(docs): libraries-bom 8.0.0 (#1066)
2020-06-26 01:34:16,384 autosynth [DEBUG] > Running: git checkout autosynth-synthtool
Switched to branch 'autosynth-synthtool'
2020-06-26 01:34:16,390 autosynth [ERROR] > Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.
2020-06-26 01:34:16,534 autosynth [DEBUG] > Running: git checkout b1353f8cc7aaf22a30b31bd2ac72248d3a3a0a5d
Note: checking out 'b1353f8cc7aaf22a30b31bd2ac72248d3a3a0a5d'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at b1353f8 chore(docs): libraries-bom 8.0.0 (#1066)
2020-06-26 01:34:16,541 autosynth [DEBUG] > Running: git checkout 652d446edabb0ea07de0ce542c6b37ab7dad3a19
HEAD is now at 652d446 chore: use GOOGLE_CLOUD_PROJECT in samples noxfile (#650)
2020-06-26 01:34:16,549 autosynth [DEBUG] > Running: git branch -f autosynth-80
2020-06-26 01:34:16,553 autosynth [DEBUG] > Running: git checkout autosynth-80
Switched to branch 'autosynth-80'
2020-06-26 01:34:16,559 autosynth [INFO] > Running synthtool
2020-06-26 01:34:16,559 autosynth [INFO] > ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']
2020-06-26 01:34:16,559 autosynth [DEBUG] > log_file_path: /tmpfs/src/logs/google-http-java-client/80/sponge_log.log
2020-06-26 01:34:16,561 autosynth [DEBUG] > Running: /tmpfs/src/github/synthtool/env/bin/python3 -m synthtool --metadata synth.metadata synth.py --
2020-06-26 01:34:16,760 synthtool [DEBUG] > Executing /home/kbuilder/.cache/synthtool/google-http-java-client/synth.py.
On branch autosynth-80
nothing to commit, working tree clean
2020-06-26 01:34:16,892 synthtool [DEBUG] > Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
DEBUG:synthtool:Using precloned repo /home/kbuilder/.cache/synthtool/synthtool
.github/CODEOWNERS
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/release-please.yml
.github/trusted-contribution.yml
.github/workflows/ci.yaml
2020-06-26 01:34:16,913 synthtool [DEBUG] > Wrote metadata to synth.metadata.
DEBUG:synthtool:Wrote metadata to synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/google-http-java-client/synth.py", line 21, in <module>
    templates = common_templates.java_library()
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 104, in java_library
    return self._generic_library("java_library", **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/gcp/common.py", line 57, in _generic_library
    result = t.render(**kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 83, in render
    _render_to_path(self.env, template_name, self.dir / subdir, kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/sources/templates.py", line 53, in _render_to_path
    output.dump(fh)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1313, in dump
    fp.writelines(iterable)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1357, in __next__
    return self._next()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 1125, in generate
    yield self.environment.handle_exception()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/environment.py", line 832, in handle_exception
    reraise(*rewrite_traceback_stack(source=source))
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/jinja2/_compat.py", line 28, in reraise
    raise value.with_traceback(tb)
  File "/home/kbuilder/.cache/synthtool/synthtool/synthtool/gcp/templates/java_library/.github/workflows/ci.yaml", line 12, in top-level template code
    java: [{% if metadata['min_java_version'] <= 7 %}7, {% endif %}8, 11]
jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'min_java_version'
2020-06-26 01:34:16,952 autosynth [ERROR] > Synthesis failed
2020-06-26 01:34:16,952 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at b1353f8 chore(docs): libraries-bom 8.0.0 (#1066)
2020-06-26 01:34:16,959 autosynth [DEBUG] > Running: git checkout autosynth
Switched to branch 'autosynth'
2020-06-26 01:34:16,965 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Traceback (most recent call last):
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 367, in synthesize_loop
    synthesize_inner_loop(fork, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 411, in synthesize_inner_loop
    synthesizer, len(toolbox.versions) - 1
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 266, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 657, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 506, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 637, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 375, in synthesize_loop
    synthesize_loop_single_pr(toolbox, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 401, in synthesize_loop_single_pr
    synthesize_inner_loop(toolbox, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 411, in synthesize_inner_loop
    synthesizer, len(toolbox.versions) - 1
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 266, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/1b8e3777-2a8a-43ca-858f-1e4587d9e4d1/targets/github%2Fsynthtool;config=default/tests;query=google-http-java-client;failed=false).

--- a/.github/CODEOWNERS
+++ b/.github/CODEOWNERS
@@ -4,5 +4,8 @@
 # For syntax help see:
 # https://help.github.com/en/github/creating-cloning-and-archiving-repositories/about-code-owners#codeowners-syntax
 
+# The @googleapis/yoshi-java is the default owner for changes in this repo
+**/*.java               @googleapis/yoshi-java
+
 # The java-samples-reviewers team is the default owner for samples changes
 samples/**/*.java       @googleapis/java-samples-reviewers

--- a/.github/CODEOWNERS
+++ b/.github/CODEOWNERS
@@ -1,1 +1,8 @@
-* @googleapis/yoshi-java
+# Code owners file.
+# This file controls who is tagged for review for any given pull request.
+
+# For syntax help see:
+# https://help.github.com/en/github/creating-cloning-and-archiving-repositories/about-code-owners#codeowners-syntax
+
+# The java-samples-reviewers team is the default owner for samples changes
+samples/**/*.java       @googleapis/java-samples-reviewers

--- a/.github/workflows/ci.yaml
+++ b/.github/workflows/ci.yaml
@@ -36,11 +36,14 @@ jobs:
         JOB_TYPE: test
   dependencies:
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        java: [8, 11]
     steps:
     - uses: actions/checkout@v2
     - uses: actions/setup-java@v1
       with:
-        java-version: 8
+        java-version: ${{matrix.java}}
     - run: java -version
     - run: .kokoro/dependencies.sh
   linkage-monitor:

--- a/.kokoro/build.sh
+++ b/.kokoro/build.sh
@@ -39,7 +39,7 @@ retry_with_backoff 3 10 \
 
 # if GOOGLE_APPLICATION_CREDIENTIALS is specified as a relative path prepend Kokoro root directory onto it
 if [[ ! -z "${GOOGLE_APPLICATION_CREDENTIALS}" && "${GOOGLE_APPLICATION_CREDENTIALS}" != /* ]]; then
-    export GOOGLE_APPLICATION_CREDENTIALS=$(realpath ${KOKORO_ROOT}/src/${GOOGLE_APPLICATION_CREDENTIALS})
+    export GOOGLE_APPLICATION_CREDENTIALS=$(realpath ${KOKORO_GFILE_DIR}/${GOOGLE_APPLICATION_CREDENTIALS})
 fi
 
 RETURN_CODE=0
@@ -51,9 +51,7 @@ test)
     RETURN_CODE=$?
     ;;
 lint)
-    mvn \
-      -Penable-samples \
-      com.coveo:fmt-maven-plugin:check
+    mvn com.coveo:fmt-maven-plugin:check
     RETURN_CODE=$?
     ;;
 javadoc)

--- a/.kokoro/dependencies.sh
+++ b/.kokoro/dependencies.sh
@@ -41,8 +41,10 @@ echo "****************** DEPENDENCY LIST COMPLETENESS CHECK *******************"
 ## Run dependency list completeness check
 function completenessCheck() {
   # Output dep list with compile scope generated using the original pom
+  # Running mvn dependency:list on Java versions that support modules will also include the module of the dependency.
+  # This is stripped from the output as it is not present in the flattened pom.
   msg "Generating dependency list using original pom..."
-  mvn dependency:list -f pom.xml -Dsort=true | grep '\[INFO]    .*:.*:.*:.*:.*' | grep -v ':test$' >.org-list.txt
+  mvn dependency:list -f pom.xml -Dsort=true | grep '\[INFO]    .*:.*:.*:.*:.*' | sed -e s/\\s--\\smodule.*// | grep -v ':test$' >.org-list.txt
 
   # Output dep list generated using the flattened pom (test scope deps are ommitted)
   msg "Generating dependency list using flattened pom..."

--- a/.kokoro/nightly/integration.cfg
+++ b/.kokoro/nightly/integration.cfg
@@ -28,14 +28,10 @@ env_vars: {
 
 env_vars: {
   key: "GOOGLE_APPLICATION_CREDENTIALS"
-  value: "keystore/73713_java_it_service_account"
+  value: "secret_manager/java-it-service-account"
 }
 
-before_action {
-  fetch_keystore {
-    keystore_resource {
-      keystore_config_id: 73713
-      keyname: "java_it_service_account"
-    }
-  }
+env_vars: {
+  key: "SECRET_MANAGER_KEYS"
+  value: "java-it-service-account"
 }

--- a/.kokoro/nightly/samples.cfg
+++ b/.kokoro/nightly/samples.cfg
@@ -24,19 +24,15 @@ env_vars: {
 
 env_vars: {
   key: "GOOGLE_APPLICATION_CREDENTIALS"
-  value: "keystore/73713_java_it_service_account"
+  value: "secret_manager/java-docs-samples-service-account"
+}
+
+env_vars: {
+  key: "SECRET_MANAGER_KEYS"
+  value: "java-docs-samples-service-account"
 }
 
 env_vars: {
   key: "ENABLE_BUILD_COP"
   value: "true"
-}
-
-before_action {
-  fetch_keystore {
-    keystore_resource {
-      keystore_config_id: 73713
-      keyname: "java_it_service_account"
-    }
-  }
 }

--- a/.kokoro/populate-secrets.sh
+++ b/.kokoro/populate-secrets.sh
@@ -1,0 +1,43 @@
+#!/bin/bash
+# Copyright 2020 Google LLC.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -eo pipefail
+
+function now { date +"%Y-%m-%d %H:%M:%S" | tr -d '\n' ;}
+function msg { println "$*" >&2 ;}
+function println { printf '%s\n' "$(now) $*" ;}
+
+
+# Populates requested secrets set in SECRET_MANAGER_KEYS from service account:
+# kokoro-trampoline@cloud-devrel-kokoro-resources.iam.gserviceaccount.com
+SECRET_LOCATION="${KOKORO_GFILE_DIR}/secret_manager"
+msg "Creating folder on disk for secrets: ${SECRET_LOCATION}"
+mkdir -p ${SECRET_LOCATION}
+for key in $(echo ${SECRET_MANAGER_KEYS} | sed "s/,/ /g")
+do
+  msg "Retrieving secret ${key}"
+  docker run --entrypoint=gcloud \
+    --volume=${KOKORO_GFILE_DIR}:${KOKORO_GFILE_DIR} \
+    gcr.io/google.com/cloudsdktool/cloud-sdk \
+    secrets versions access latest \
+    --project cloud-devrel-kokoro-resources \
+    --secret ${key} > \
+    "${SECRET_LOCATION}/${key}"
+  if [[ $? == 0 ]]; then
+    msg "Secret written to ${SECRET_LOCATION}/${key}"
+  else
+    msg "Error retrieving secret ${key}"
+  fi
+done

--- a/.kokoro/presubmit/integration.cfg
+++ b/.kokoro/presubmit/integration.cfg
@@ -24,14 +24,10 @@ env_vars: {
 
 env_vars: {
   key: "GOOGLE_APPLICATION_CREDENTIALS"
-  value: "keystore/73713_java_it_service_account"
+  value: "secret_manager/java-it-service-account"
 }
 
-before_action {
-  fetch_keystore {
-    keystore_resource {
-      keystore_config_id: 73713
-      keyname: "java_it_service_account"
-    }
-  }
+env_vars: {
+  key: "SECRET_MANAGER_KEYS"
+  value: "java-it-service-account"
 }

--- a/.kokoro/presubmit/samples.cfg
+++ b/.kokoro/presubmit/samples.cfg
@@ -24,14 +24,10 @@ env_vars: {
 
 env_vars: {
   key: "GOOGLE_APPLICATION_CREDENTIALS"
-  value: "keystore/73713_java_it_service_account"
+  value: "secret_manager/java-docs-samples-service-account"
 }
 
-before_action {
-  fetch_keystore {
-    keystore_resource {
-      keystore_config_id: 73713
-      keyname: "java_it_service_account"
-    }
-  }
+env_vars: {
+  key: "SECRET_MANAGER_KEYS"
+  value: "java-docs-samples-service-account"
 }

--- a/.kokoro/trampoline.sh
+++ b/.kokoro/trampoline.sh
@@ -21,4 +21,6 @@ function cleanup() {
     echo "cleanup";
 }
 trap cleanup EXIT
+
+$(dirname $0)/populate-secrets.sh # Secret Manager secrets.
 python3 "${KOKORO_GFILE_DIR}/trampoline_v1.py"

--- a/.repo-metadata.json
+++ b/.repo-metadata.json
@@ -6,5 +6,6 @@
   "language": "java",
   "repo": "googleapis/google-http-java-client",
   "repo_short": "google-http-java-client",
+  "codeowner_team": "@googleapis/yoshi-java",
   "distribution_name": "com.google.http-client:google-http-client"
 }

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -99,7 +99,16 @@ mvn -Penable-samples clean verify
     ```
 
 2. [Activate](#profile-activation) the profile.
-3. Define your samples in a normal Maven project in the `samples/` directory
+3. Define your samples in a normal Maven project in the `samples/` directory.
+
+### Code Formatting
+
+Code in this repo is formatted with
+[google-java-format](https://github.com/google/google-java-format).
+To run formatting on your project, you can run:
+```
+mvn com.coveo:fmt-maven-plugin:format
+```
 
 ### Profile Activation
 

--- a/synth.metadata
+++ b/synth.metadata
@@ -3,15 +3,15 @@
     {
       "git": {
         "name": ".",
-        "remote": "https://github.com/googleapis/google-http-java-client.git",
-        "sha": "1624d55f9864ff1253321e9bfe715e2f370cc27e"
+        "remote": "git@github.com:chingor13/google-http-java-client.git",
+        "sha": "b1353f8cc7aaf22a30b31bd2ac72248d3a3a0a5d"
       }
     },
     {
       "git": {
         "name": "synthtool",
         "remote": "https://github.com/googleapis/synthtool.git",
-        "sha": "98c50772ec23295c64cf0d2ddf199ea52961fd4c"
+        "sha": "ce68c0e70d36c93ffcde96e9908fb4d94aa4f2e4"
       }
     }
   ]

--- a/synth.py
+++ b/synth.py
@@ -12,14 +12,10 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 """This script is used to synthesize generated parts of this library."""
-import synthtool as s
-import synthtool.gcp as gcp
-import logging
 
-logging.basicConfig(level=logging.DEBUG)
-common_templates = gcp.CommonTemplates()
-templates = common_templates.java_library()
-s.copy(templates, excludes=[
+import synthtool.languages.java as java
+
+java.common_templates(excludes=[
     "README.md",
     "java.header",
     "checkstyle.xml",

